src/providers/helpers/uploadObservation.js

```diff
@@ -0,0 +1,90 @@
+// @flow
+
+import inatjs from "inaturalistjs";
+
+import { getJWTToken } from "../../components/LoginSignUp/AuthenticationService";
+import { markUploaded, markPhotoUploaded, markSoundUploaded } from "./markUploaded";
+import ObservationPhoto from "../../models/ObservationPhoto";
+import ObservationSound from "../../models/ObservationSound";
+// import fetchPlaceName from "../../../sharedHelpers/fetchPlaceName";
+
+const uploadSound = async ( soundParams, apiToken, uuid ) => {
+  const options = { api_token: apiToken };
+
+  try {
+    const { results } = await inatjs.observation_sounds.create( soundParams, options );
+    const soundId = results[0].id;
+    console.log( soundId, "sound created" );
+    await markSoundUploaded( uuid, soundId );
+  } catch ( e ) {
+    console.log( JSON.stringify( e.response ), "couldn't upload sound" );
+  }
+};
+
+const createSoundParams = async ( id, apiToken, localObs ) => {
+  const obsSoundToUpload = localObs.observationSound;
+  const soundParams = ObservationSound.mapSoundForUpload( id, obsSoundToUpload );
+  console.log( soundParams, "sound params" );
+  uploadSound( soundParams, apiToken, obsSoundToUpload.uuid );
+};
+
+const uploadPhoto = async ( photoParams, apiToken, uuid ) => {
+  const options = { api_token: apiToken };
+
+  try {
+    const { results } = await inatjs.observation_photos.create( photoParams, options );
+    const photoId = results[0].id;
+    await markPhotoUploaded( uuid, photoId );
+  } catch ( e ) {
+    console.log( JSON.stringify( e.response ), "couldn't upload photo" );
+  }
+};
+
+const createPhotoParams = async ( id, apiToken, localObs ) => {
+  const obsPhotosToUpload = localObs.observationPhotos;
+
+  if ( !obsPhotosToUpload || obsPhotosToUpload.length === 0 ) { return; }
+  for ( let i = 0; i < obsPhotosToUpload.length; i += 1 ) {
+    const photoToUpload = obsPhotosToUpload[i];
+    const photoParams = ObservationPhoto.mapPhotoForUpload( id, photoToUpload );
+    uploadPhoto( photoParams, apiToken, photoToUpload.uuid );
+  }
+};
+
+const uploadObservation = async ( obsToUpload: Object, localObs: Object ) => {
+  const FIELDS = { id: true };
+
+  try {
+    const apiToken = await getJWTToken( false );
+
+    const uploadParams = {
+      // TODO: decide how to format place_guess param
+      // right now it looks like street, city, state is preferred on the web
```

Comment:
Formatting the `place_guess` should probably not happen during upload. That's user-editable, so any auto-population of that field should happen when the coordinates change.